### DISCUSSION TITLE
Usecase Include? precondition?

### DISCUSSION THREAD

**Discussion by minseungseon** [cau-swe/checkpoint-1#16] [General]
usecase 중간에 어떤 작업에 대해서 다른 usecase 에서도 해당 작업이 동일하게 반복되면 include 를 사용한다. 
처음부터 어느 지점까지 동일한 작업이면 precondition을 쓴다!

**Comment by Topslug** [comment]
민승님이 잘 정리해서 올려주셨지만, 카톡에 올렸던 내용을 여기에 다시 게시해 놓겠습니다.

Include와 precondition의 차이에 관한 저의 생각입니다.

a b c d e f g라는 flow of Events가 있을 때(a는 모든 행위의 첫 시작, 각 알파벳은 하나의 행위를 나타낸다.)
1. 만약 b c d라는 행위가 다른 use case에도 계속해서 반복되면 b c d를 다른 Use case(ex- A)로 이름 붙인 후
a include A e f g 이런 식으로 flow of Events를 작성한다고 생각합니다. (Use Case 1 : Unlock, Use Case 9 : ManageUsers)
2. 만약 a b c d를 수행해야만 e f g가 수행이 될 수 있고 다른 use case에도 a b c d가 반복된다고 할 때, a b c d를 Use Case A라고 하고
precondition에 A까지 된 상태라고 언급 후
e f g를 Flow of Events에 쓰는 것이 맞는 것 같습니다. (Use Case 5 : Inspect Access History)
혹은 Flow of Events : include A e f g이렇게 써도 될 것 같습니다.

Use Case 9 : ManageUsers를 보면 ManageUsers버튼을 누르는 것은 분명 Login뒤일텐데(include Login을 쓰고 selects ManageUsers를 쓰는 것이 순서 상으로는 맞지만) 그냥 selects ManageUsers를 쓰고 include Log in을 한 것으로 봐서, 본인 기호에 맞는 표현을 아무거나 쓰면 되는 것 같기도 합니다.

**Comment by Topslug** [comment]
그리고 창선님께서 교수님께 질문 드렸던 내용도 여기 게시하겠습니다.

<img width="1403" alt="KakaoTalk_20210412_162322601" src="https://user-images.githubusercontent.com/53007295/114976313-b6a35900-9ec0-11eb-9cd6-6b36a429b41f.png">
<img width="1412" alt="KakaoTalk_20210412_162749600" src="https://user-images.githubusercontent.com/53007295/114976320-b99e4980-9ec0-11eb-8f96-a384c40a6eaf.png">
